File: erlang/sum-of-multiples/src/sum_of_multiples.erl
-module(sum_of_multiples).
-export([sum/2]).
-spec sum(list(), integer()) -> integer().
sum(Factors, Limit) ->
  aux(Factors, Limit - 1, 0).
aux(_Factors, 0, Sum) ->
  Sum;
aux(Factors, Limit, Sum) ->
  IsMultiple = lists:any(fun(0) -> false;
                            (I) -> Limit rem I == 0 end,
                         Factors),
  case IsMultiple of
    true  -> aux(Factors, Limit - 1, Sum + Limit);
    false -> aux(Factors, Limit - 1, Sum)
  end.

File: elixir/sum-of-multiples/lib/sum_of_multiples.ex
defmodule SumOfMultiples do

  def to(limit, factors) do
    1..(limit - 1)
    |> Enum.filter(fn n -> factors |> Enum.any?(&is_multiple?(n, &1)) end)
    |> Enum.sum()
  end
  
  defp is_multiple?(_, 0), do: false
  defp is_multiple?(num, factor), do: rem(num, factor) == 0
  
end
